Sources/MuxPlayerSwift/InternalExtensions/AVPlayerItem+Mux.swift

```diff
@@ -113,3 +148,363 @@ internal extension AVPlayerItem {
+                
+                do {
+                    try await answerRequestForMediaPlaylist(
+                        resourceLoadingRequest: loadingRequest,
+                        playlistURL: url,
+                        originBaseURL: makeOriginBaseURL(playlistURL: url),
+                        cacheBaseURL: URL(string: "https://mux.com")! // TODO: Didn't need this param
+                    )
+                } catch {
+                    PlayerSDK.shared.diagnosticsLogger.error(
+                        "Error fetching/generating short-form playlist: \(error.localizedDescription)"
+                    )
+                    loadingRequest.finishLoading(with: error)
+                }
+                fetchTask = nil
+            }
+            
+            return true
+        } else {
+            return false
+        }
+    }
+    
+    func resourceLoader(
+        _ resourceLoader: AVAssetResourceLoader,
+        didCancel loadingRequest: AVAssetResourceLoadingRequest
+    ) {
+        // As long as the methods of this delegate are called without interleaving (i think they are), this should be fine
+        fetchTask?.cancel()
+    }
+    
+    private func answerRequestForMediaPlaylist(
+        resourceLoadingRequest loadingRequest: AVAssetResourceLoadingRequest,
+        playlistURL: URL,
+        originBaseURL: URL,
+        cacheBaseURL: URL
+    ) async throws {
+        // TODO: The most straightforward way to cache the init segment is via the same proxy cache as everything else, but this *does* mean we might evict an init segment and have to re-request it even in the same app session (which is probably not necessary)
+        //        let initSegmentURL = makeInitSegmentURL(playlistURL: playlistURL)
+        let initSegmentURL = makeCacheProxyURL(forFullURL: makeInitSegmentURL(playlistURL: playlistURL))
+        
+        PlayerSDK.shared.diagnosticsLogger.info(
+            "[shorform-test] initSegmentURL: \(initSegmentURL.absoluteString)"
+        )
+
+        let segmentData = try await fetchInitSegment(initSegmentURL: initSegmentURL)
+        PlayerSDK.shared.diagnosticsLogger.debug("resourceLoader fetched \(segmentData.count) bytes")
+        
+        // TODO: The target duration (and real segment duration; can be different) should maybe come down from the server via Response Header, if that's the way the origin is going to give us information
+        // TODO: (continued) .. although for playback, we might not need the real target duration...
+        let playlistString = try ShortFormMediaPlaylistGenerator(
+            initSegment: segmentData,
+            originBaseURL: originBaseURL,
+            cacheProxyBaseURL: cacheBaseURL,
+            playlistAttributes: ShortFormMediaPlaylistGenerator.PlaylistAttributes(
+                version: 7,
+//                        targetDuration: 5, // TODO: wouldn't a mux video asset have 6?
```

Comment:
these can vary fyi

Reply:
Ahh gotcha. Might be good for these to come down via response header then?